src/rgw/rgw_op.cc

```diff
@@ -3951,6 +3951,7 @@ void RGWPutObj::execute(optional_yield y)
 
     s->trace->SetAttribute(tracing::rgw::UPLOAD_ID, multipart_upload_id);
     multipart_trace = tracing::rgw::tracer.add_span(name(), upload->get_trace());
+    s->object->set_trace(multipart_trace->GetContext());
```

Comment:
lets start from regular object trace. please move out of the "if multipart" block, and set the regular trace as its parent